File: Traitement d'une partition/C++/src/effacementPortee.cpp
#include <effacementPortee.h>

void effacementPortee(Eigen::MatrixXi & imageCorrigee, Eigen::VectorXi Py, int interligneMoyen, Eigen::VectorXi localisation,
Eigen::VectorXi offsetX, Eigen::VectorXi Hprime, int ydebut, int yfin, int emoy){
    int N = localisation.rows();
    int se = emoy + 3;
    for(int y = ydebut ; y < yfin ; y++){
        for(int i = 0 ; i < N ; i++){
            int xh = localisation(i);
            int xb = localisation(i);
            int x = localisation(i);
            while(imageCorrigee(xb, y) == 1){
                xb++;
            }while(imageCorrigee(xh, y) == 1){
                xh--;
            }if((se>= (xb-xh+1)) && (xb < (x+se)) && (xh>(x-se))){
                for(int i = xh-2 ; i <= xb ; i++)
                imageCorrigee(i, y) = 0;
            }
        }
    }/*
    int xm = imageCorrigee.rows();
    int ym = imageCorrigee.cols();
    for(int x = 0 ; x < xm ; x++){
        for(int y = 0 ; y < ym ; y++){
            imageCorrigee(x,y) = 0;
        }
    }*/
}

File: Traitement d'une partition/Mathlab/Source/effacementportee.m
% effacement des portées
% 
 function[imagesansportee] = effacementportee(im_corr,localisation,ydebut,yfin,emoy)

% %données tests
% close all
% clear all
% im_corr=correction_biais(imread('Jeregretteun.bmp'));
% [H, W]=size(im_corr);
% [Py,interligne_moy] = detectioninterligne(im_corr); %im_corr c'est de 0 (noir et des 1 blanc)
% localisation = rechechedesxis(Py, W, interligne_moy);
% [abscisse_troisieme_ligne, offsetX, Hp, ox] = sous_image_portee(localisation, im_corr, interligne_moy);

image_corr=1-im_corr;


se=floor(emoy)+3;
x=0;
[C B]=size(image_corr)
for y=ydebut:yfin
    for x=localisation
        xh=x;
        xb=x;
                %détermination de xb
                while (image_corr(xb,y)==1 && xb<C)
                    xb=xb+1;
                end
                %détermination de xh
                while (image_corr(xh,y)==1 && xh<C)
                    xh=xh-1;
                end
                A=(se>= (xb-xh+1)) && (xb < (x+se)) && (xh>(x-se));
        if (A)
            image_corr(xh-1:xb+1,y)=0;
        end
    end
end
imagesansportee=image_corr;
% imshow(1-image_corr)
% title('image corrigée')